Apps/EGIAmpServer/EGIAmpWindow.cpp

```diff
@@ -31,236 +119,691 @@ EGIAmpWindow::EGIAmpWindow(QWidget* parent, const std::string& config_file)
 	                 &EGIAmpWindow::load_config_dialog);
 	QObject::connect(ui->actionSave_Configuration, &QAction::triggered, this,
 	                 &EGIAmpWindow::save_config_dialog);
+	QObject::connect(this, &EGIAmpWindow::appendStatusMessage, ui->statusBox,
+	                 &QPlainTextEdit::appendPlainText);
+	QObject::connect(this, &EGIAmpWindow::enableSampleRateSelection, ui->sampleRateComboBox,
+	                 &QComboBox::setEnabled);
+	QObject::connect(this, &EGIAmpWindow::error, this, &EGIAmpWindow::displayError);
+	QObject::connect(this, &EGIAmpWindow::enableUI, this, &EGIAmpWindow::unlockUI);
+	QObject::connect(this, &EGIAmpWindow::disableUI, this, &EGIAmpWindow::lockUI);
+	QObject::connect(this, &EGIAmpWindow::fieldsEnabled, this->ui->sampleRateComboBox,
+	                 &QComboBox::setEnabled);
+	QObject::connect(this, &EGIAmpWindow::fieldsEnabled, this->ui->amplifierId,
+	                 &QSpinBox::setEnabled);
+	QObject::connect(this, &EGIAmpWindow::fieldsEnabled, this->ui->serverAddress,
+	                 &QLineEdit::setEnabled);
+	QObject::connect(this, &EGIAmpWindow::fieldsEnabled, this->ui->commandPort,
+	                 &QSpinBox::setEnabled);
+	QObject::connect(this, &EGIAmpWindow::fieldsEnabled, this->ui->notificationPort,
+	                 &QSpinBox::setEnabled);
+	QObject::connect(this, &EGIAmpWindow::fieldsEnabled, this->ui->dataPort, &QSpinBox::setEnabled);
+	QObject::connect(this, &EGIAmpWindow::setLinkButtonText, this->ui->linkButton,
+	                 &QPushButton::setText);
 }
 
 void EGIAmpWindow::load_config_dialog() {
-	QString sel = QFileDialog::getOpenFileName(this,"Load Configuration File","","Configuration Files (*.cfg)");
-	if (!sel.isEmpty())
-		load_config(sel.toStdString());
+	QString sel = QFileDialog::getOpenFileName(this, "Load Configuration File", "",
+	                                           "Configuration Files (*.cfg)");
+	if (!sel.isEmpty()) load_config(sel.toStdString());
 }
 
 void EGIAmpWindow::save_config_dialog() {
-	QString sel = QFileDialog::getSaveFileName(this,"Save Configuration File","","Configuration Files (*.cfg)");
-	if (!sel.isEmpty())
-		save_config(sel.toStdString());
+	QString sel = QFileDialog::getSaveFileName(this, "Save Configuration File", "",
+	                                           "Configuration Files (*.cfg)");
+	if (!sel.isEmpty()) save_config(sel.toStdString());
 }
 
-void EGIAmpWindow::closeEvent(QCloseEvent *ev) {
-	if (reader_thread_)
-		ev->ignore();
+void EGIAmpWindow::closeEvent(QCloseEvent* ev) {
+	if (reader_thread_) ev->ignore();
 }
 
-void EGIAmpWindow::load_config(const std::string &filename) {
+void EGIAmpWindow::load_config(const std::string& filename) {
```

Comment:
Thank you. 

On a related note, I eliminated the superfluous thread (mentioned below), but did not reformat the segment of code to adjust indentation because doing so obfuscated the changes represented in this update. I assumed that if we are going to reformat the code with the clang-format configuration, this would take care of the indentation and the advantage of limiting the changeset to those things that actually changed was worth leaving the indentation as is. If you'd like me to adjust the indentation and commit this change I can do so.